### PR TITLE
salt: Implement ping_minions_{serial,parallel} methods

### PR DESCRIPTION
**Pull request by smithfarm**
Still to do:

* figure out how to populate `run` property
* set `run` grain to the value of the `run` property (do this in `init_minions()`)
* document the class itself (including usage example)

### CODE REVIEW

**Comment by jan--f** [comment]
looking good @smithfarm. testing your branch right now.

**Comment by smithfarm** [comment]
@jan--f Great, thanks. Before we merge we should run `tox -e py27` (forgot to do that before merging #19)

File: teuthology/salt.py [hunk 1]
```diff
@@ -53,9 +53,9 @@ def __init__(self, ctx, config, **kwargs):
         self._teuthology_ip_address = None
         self._minions = []
         if 'master_fqdn' not in kwargs:
-            self.master_fqdn(self.teuthology_fqdn)
+            self._master_fqdn(self.teuthology_fqdn)
         else:
-            self.master_fqdn(kwargs['master_fqdn'])
+            self._master_fqdn(kwargs['master_fqdn'])
```
Comment (smithfarm) [inline teuthology/salt.py]: I thought I didn't need the leading underscore if I defined a setter called `master_fqdn` (see line 83)

Comment (jan--f) [inline teuthology/salt.py]: yes the setter is defined on a Salt object. In the class contructor you need to set the actual property, the setter exposes.

File: teuthology/salt.py [hunk 2]
```diff
@@ -90,22 +95,12 @@ def minions(self):
...
         """Return the resolvable FQDN of the teuthology VM"""
-        ip_addr = self.teuthology_ip_address.split('.')
-        log.debug("teuthology_ip_address returned {}".format(ip_addr))
-        return "target{:03d}{:03d}{:03d}{:03d}.teuthology".format(
-            int(ip_addr[0]),
-            int(ip_addr[1]),
-            int(ip_addr[2]),
-            int(ip_addr[3]),
-        )
+        return self._teuthology_fqdn
```
Comment (smithfarm) [inline teuthology/salt.py]: nice!

File: teuthology/salt.py [hunk 3]
```diff
@@ -234,7 +239,7 @@ def ping_minions_serial(self):
     def ping_minions_parallel(self):
         """Pings minions, raises exception if they don't respond"""
         if self.master_fqdn == self.teuthology_fqdn:
-            sh("sudo salt -C 'G@run:{}' test.ping".format(self.run))
+            sh("sudo salt -C 'G@job_id:{}' test.ping".format(self.job_id))
         else:
```
Comment (smithfarm) [inline teuthology/salt.py]: :thumbsup:

**Pull request merged**